python/src/api_media_object.py

```diff
@@ -0,0 +1,66 @@
+import re
+
+import requests
+
+from api_object import ApiObject
+
+
+class ApiMediaObject(ApiObject):
+    """
+    Subclass of an ApiObject with media functionality.
+    """
+
+    def __init__(self, api_config, access_token):
+        super().__init__(api_config, access_token)
+
+    def upload_media(self, media):
+        """
+        The implementation of this function depends on the API version,
+        so it must be overridden in the subclass for each version of the API.
+        """
+        raise RuntimeError("upload_media() must be overridden")
+
+    def media_to_media_id(self, media):
+        """
+        This function translates the media argument into a media_id,
+        which may be one of:
+          <falsy>     => no video creation is required
+          <file path> => create a media_id from the video in the file path
+          media_id    => an existing media identifier
+
+        References:
+          v3: https://developers.pinterest.com/docs/redoc/#section/Using-video-APIs
+          v5: https://developers.pinterest.com/docs/solutions/content-apps/#creatingvideopins
+        """  # noqa: E501 because the long URL is okay
+        if not media:
+            return media
+
+        try:  # check whether media is a readable file path
+            open(media, "r").close()
+        except OSError:
+            # media is not a readable file path. check whether it is a valid media_id
+            if re.match(r"\d+$", media):
```

Comment:
You could use [`str.isdigit`](https://docs.python.org/3.10/library/stdtypes.html#str.isdigit) here:

```suggestion
            if media.isdigit():
```

Reply:
thanks!